src/core/did.ts

```diff
@@ -0,0 +1,69 @@
+import { Id } from './id';
+
+const method = 'iden3';
+
+export enum Blockchain {
+  Ethereum = 'eth',
+  Polygon = 'polygon',
+}
+
+export enum NetworkId {
+  Main = 'main', // mainnet
+  Test = 'test', // testnet
+  Roptsen = 'ropsten',
+  Rinkeby = 'rinkeby',
+  Kovan = 'kovan',
+  Mumbai = 'mumbai'
+}
+const didRegex = new RegExp(
+  `^did:iden3:((eth|polygon):(main|test|ropsten|rinkeby|kovan):)?[1-9a-km-zA-HJ-NP-Z]{41,42}$`
+);
+const errDoesNotMatchRegexp = new Error('did does not match regex');
+
+// DID Decentralized Identifiers (DIDs)
+// https://w3c.github.io/did-core/#did-syntax
+export class DID {
+  id: Id; // Id did specific id
+  blockchain: Blockchain; // Blockchain network identifier eth / polygon,...
+  networkId: NetworkId; // NetworkId specific network identifier eth {main, ropsten, rinkeby, kovan}
+  constructor(idStr: string, blockchain?: Blockchain, networkId?: NetworkId) {
```

Comment:
add access modifiers to blockchain and networkId and remove line 27 and 28

Reply:
fixed